conjure-core/src/main/java/com/palantir/conjure/defs/validator/FieldNameValidator.java

```diff
@@ -53,7 +56,7 @@ public static void validate(FieldName fieldName) {
                         || CaseConverter.SNAKE_CASE_PATTERN.matches(fieldName.get()),
                 "FieldName \"%s\" must follow one of the following patterns: %s",
                 fieldName,
-                Arrays.toString(CaseConverter.Case.values()));
+                CASE_VALUES);
```

Comment:
instead of precomputing the components of the failure message, what if we simplify this to replace the preconditions check with a simple conditional? This way the error message components are only accessed when needed, and all the relevant data lives in the same place.

In most executions, the precomputed `Arrays.toString(Case.values())` isn't read, so it's wasted computation (albeit small and executed once).